packages/core/utils/lib/sanitize/index.js

```diff
@@ -37,12 +37,16 @@ const createContentAPISanitizers = () => {
     return pipeAsync(...transforms)(data);
   };
 
-  const sanitizeOuput = (data, schema, { auth } = {}) => {
+  const sanitizeOuput = async (data, schema, { auth } = {}) => {
     if (isArray(data)) {
-      return Promise.all(data.map((entry) => sanitizeOuput(entry, schema, { auth })));
+      const res = [];
+      for (let i = 0; i < data.length; i += 1) {
+        res.push(await sanitizeOuput(data[i], schema, { auth }));
+      }
+      return res;
     }
```

Comment:
maybe it doesnt add much, but according to this blog https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/

If ou preallocate memory on the res array, its more performant:
```suggestion
      const res = new Array(data.length);
      for (let i = 0; i < data.length; i += 1) {
        res.push(await sanitizeOuput(data[i], schema, { auth }));
      }
      return res;
    }
```

Reply:
Interesting. we could try that. Probably not gonna make a diff at this stage but good to know when we have to go down to micro optimizations

Reply:
It's actually quite a bit of performance increase (12%) according to [perflink](https://perf.link/#eyJpZCI6Imd5MWtxY3g4Yzh4IiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGRhdGEgPSBbLi4uQXJyYXkoMTAwMCkua2V5cygpXSIsInRlc3RzIjpbeyJuYW1lIjoiRmluZCBpdGVtIDEwMCIsImNvZGUiOiIgY29uc3QgcmVzID0gbmV3IEFycmF5KGRhdGEubGVuZ3RoKTtcbiAgZm9yIChsZXQgaSA9IDA7IGkgPCBkYXRhLmxlbmd0aDsgaSArPSAxKSB7XG4gICAgcmVzW2ldID0xO1xuICB9IiwicnVucyI6WzUwMDAsNTQwMDAsMjIwMDAsNTYwMDAsMTAwMCwxMDAwLDcxMDAwLDExMDAwLDcwMDAwLDE4MDAwLDc2MDAwLDEyMDAwLDE3MDAwLDIyMDAwLDU2MDAwLDcwMDAsMjkwMDAsMjUwMDAsMTgwMDAsMTAwMDAsNzkwMDAsMTEwMDAsNTAwMCw2OTAwMCw4MDAwLDU4MDAwLDExMDAwLDMwMDAsNjAwMCwyOTAwMCwyMDAwMCwxMTAwMCwzMTAwMCwyNzAwMCw2NTAwMCwzMDAwLDQxMDAwLDIwMDAsMzkwMDAsMTMwMDAsMjMwMDAsNDAwMCwxMDAwMCwxOTAwMCwyODAwMCw0MzAwMCwxNjAwMCwxNTAwMCwxNjAwMCw3MjAwMCwxOTAwMCwxMzAwMCwyNTAwMCw2NjAwMCw1OTAwMCw3MDAwLDMzMDAwLDE2MDAwLDIzMDAwLDkwMDAsMjIwMDAsMTAwMCw0MDAwMCwzMzAwMCwxMTAwMCwxMTAwMCwyMzAwMCwxNTAwMCwzMDAwMCw2MDAwLDQyMDAwLDEzMDAwLDcwMDAsNDEwMDAsOTAwMCwzNjAwMCwxODAwMCw0MTAwMCwxMTAwMCw2ODAwMCwxNTAwMCwxNzAwMCwxMDAwMCwyMzAwMCw1MDAwLDMxMDAwLDExMDAwLDExMDAwLDExMDAwLDI1MDAwLDcwMDAsNTYwMDAsMTEwMDAsMjEwMDAsMTcwMDAsMTEwMDAsNTIwMDAsMzgwMDAsMTMwMDAsMTEwMDBdLCJvcHMiOjI1MDIwfSx7Im5hbWUiOiJGaW5kIGl0ZW0gMjAwIiwiY29kZSI6IiBjb25zdCByZXMgPSBbXTtcbiAgZm9yIChsZXQgaSA9IDA7IGkgPCBkYXRhLmxlbmd0aDsgaSArPSAxKSB7XG4gICAgcmVzLnB1c2goMSk7XG4gIH0iLCJydW5zIjpbMjkwMDAsMjgwMDAsMTMwMDAsMTEwMDAsMjgwMDAsMjAwMCwzNTAwMCw1MDAwLDQxMDAwLDEwMDAsMzIwMDAsNzAwMCw0NzAwMCwyMDAwLDMyMDAwLDQ0MDAwLDEwMDAwLDE5MDAwLDcwMDAsNDMwMDAsNDUwMDAsNDcwMDAsNDcwMDAsMzMwMDAsMzYwMDAsMzYwMDAsMzAwMCwzNjAwMCw0MDAwLDcwMDAsNDAwMCwzMDAwLDQ5MDAwLDIwMDAsNDAwMDAsNDEwMDAsMjEwMDAsMzYwMDAsNDcwMDAsNzAwMCwxNTAwMCwyMDAwLDQwMDAsNDEwMDAsNTAwMCwyMTAwMCw1MDAwLDcwMDAsNTAwMCwxMDAwLDYwMDAsOTAwMCw0MjAwMCwzNjAwMCwzOTAwMCw0MTAwMCw3MDAwLDYwMDAsMTcwMDAsMzgwMDAsODAwMCwzOTAwMCwxMjAwMCwxODAwMCw0NjAwMCw0NDAwMCwzMDAwLDQ3MDAwLDM5MDAwLDQwMDAwLDIwMDAwLDQzMDAwLDQyMDAwLDI1MDAwLDQ0MDAwLDgwMDAsNzAwMCwxNTAwMCwzNzAwMCw0MjAwMCw0MDAwLDMwMDAsMzIwMDAsMTIwMDAsMzIwMDAsMTEwMDAsMzAwMCwzODAwMCwxNTAwMCw4MDAwLDQ0MDAwLDM1MDAwLDIwMDAsNzAwMCwyMjAwMCw2MDAwLDE2MDAwLDIyMDAwLDQ2MDAwLDUwMDBdLCJvcHMiOjIyNjkwfV0sInVwZGF0ZWQiOiIyMDIzLTAzLTE3VDA3OjQ1OjM0LjA4MFoifQ%3D%3D)

Reply:
the more elements the array has the more performance increase though, for 1000 elements is a 10%, for 10k 30%

Reply:
Hey fill free to push to this branch to test this technique 👍

Reply:
if you know the array length wouldn't it be easier to just reassign it? Not sure if it makes a difference @Marc-Roig

Reply:
we try not to mutate the original data I guess, would it be a problem here? @alexandrebodin

Reply:
when the delete object is executed the original object has been copied (inside`traverseEntity`), at this point , in the `sanitizeOutput` I am not so sure we can do that, as we could be mutating an objec that is used in other async processes such as webhooks.

But that is only my guess.

Reply:
Ok just tried both ways, it reduces 50ms (avg) from 1s response either way.